docs/design-proposals/node/Partial-Runtime-Readiness-Handling.md

```diff
@@ -9,20 +9,40 @@ ready or notReady. As a result, the node status and node readiness management ne
 situations on the node, and the pod placement logic at the scheduler needs to be changed to ensure POD was scheduled to 
 a node with desired runtime service.
 
+Initially Arktos is currently implemented as "NodeReady requires ANY one of its runtime services ready" for best resource 
+utilization, i.e. the scheduler can schedule the pod to a node as long as its desired runtime service ready on the node.
+However, this implementation carries a drawback with PODs with cross workload type dependencies which can be described
+as followings:
+
+A cluster is setup and supports two runtime services, containerD and Arktos VM runtime,  and their corresponding workload
+types, i.e. Container workload type and VM workload type. The network agent, Alcor agent, was deployed as a cluster daemonset,
+which is running as container workload. All tenant level workload PODs, either containers or VMs will need Alcor's agent in
+order to have its network up. A pod can be scheduled to a node ONLY VM runtime ready. However, the pod cannot run because
+the network agent failed to setup the network for it.
+ 
+The following diagram illustrates the node status dependency, and cross workload type pod dependencies.
```

Comment:
"cross workload type pod dependencies" is very vague and hard to detect in our system.  You can simply say we require container runtime to always be deployed because most cluster services are deployed as containers. And we don't want to force users to change them to vm-based YAML. Alcor agent is one example.

Reply:
reworded the use case for clarity. i guess your comment have been stated in the "Primary runtime of a cluster" section when it explains why the container runtime as default primary in our implmentation.